astropy/io/misc/tests/test_hdf5.py

```diff
@@ -567,6 +567,28 @@ def test_read_h5py_objects(tmpdir):
     f.close()         # don't raise an error in 'test --open-files'
 
 
+@pytest.mark.skipif('not HAS_H5PY')
+def test_write_to_Hdf5(tmpdir):
+    test_file = str(tmpdir.join('test.hdf5'))
+    t1 = Table()
+    t1['x']=['a', 'b', 'c']
+    t1['y']=[ 1, 2, 3 ]
+    t1['z']=[b'a', b'b', b'c']
+    t1.write(test_file, path='the_table', overwrite=True)
+    f = h5py.File(test_file)
+    t2 = Table.read(f, path='the_table')
+    t = Table.read(test_file, path='the_table')
+    assert np.all(t1['x'] == ['a', 'b', 'c'])
+    assert np.all(t1['y'] == [ 1, 2 , 3 ])
+    assert np.all(t1['z'] == [b'a', b'b', b'c'])
+    assert np.all(t['x'] == ['a', 'b', 'c'])
+    assert np.all(t['y'] == [ 1 ,2 ,3])
+    assert np.all(t['z'] == [b'a', b'b', b'c'])
+    assert np.all(t2['x'] == ['a', 'b', 'c'])
+    assert np.all(t2['y'] == [ 1, 2 , 3 ])
+    assert np.all(t2['z'] == [b'a', b'b', b'c'])
```

Comment:
Please also check the dtype of the different columns - I suspect that t2['x'] is actually still going to be a bytes column not unicode.

Reply:
Since Astropy 2.0 bytes columns in Table are handled transparently with a "unicode sandwich" (http://docs.astropy.org/en/latest/table/access_table.html#bytestring-columns-in-python-3) so you should be able to read the file and compare directly the values to the original unicode ones.
``In [3]: t = Table.read('test.hdf5', path='the_table')

In [4]: t
Out[4]: 
<Table length=3>
  x   
bytes1
------
     a
     b
     c

In [5]: t['x'] == ['a', 'b', 'c']
Out[5]: array([ True,  True,  True])``
The columns will be bytes columns but that should not be a problem.